src/Altinn.Notifications.Integrations/Authorization/AuthorizationClient.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System.Security.Claims;
+
+using Altinn.Authorization.ABAC.Xacml.JsonProfile;
+using Altinn.Common.PEP.Constants;
+using Altinn.Common.PEP.Helpers;
+using Altinn.Common.PEP.Interfaces;
+using Altinn.Notifications.Core.Integrations;
+using Altinn.Notifications.Core.Models.ContactPoints;
+using static Altinn.Authorization.ABAC.Constants.XacmlConstants;
+
+namespace Altinn.Notifications.Integrations.Authorization;
+
+/// <summary>
+/// An implementation of <see cref="IAuthorizationService"/> able to check that a potential
+/// recipient of a notification can access the resource that the notification is about.
+/// </summary>
+public class AuthorizationClient : IAuthorizationService
```

Comment:
what about calling it a Service rather than client. The client is implemented and used within PDP I would say, this is just a service using yet another service.